pytorch_lightning/plugins/training_type/full_sharded.py

```diff
@@ -0,0 +1,175 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Optional
+
+import torch
+
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.plugins.training_type.ddp import DDPPlugin
+from pytorch_lightning.utilities import _FAIRSCALE_AVAILABLE, _FAIRSCALE_FULL_SHARDED_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _FAIRSCALE_AVAILABLE:
+    from fairscale.nn.data_parallel.fully_sharded_data_parallel import (
+        FullyShardedDataParallel, Parameter, TrainingState)
+
+    from pytorch_lightning.overrides.fairscale import (
+        LightningFullShardedDataParallel,
+        unwrap_lightning_module_full_sharded,
+    )
+
+
+class LightningFullyShardedDataParallel(FullyShardedDataParallel):
```

Comment:
Is there some option, where we can either rely on the original Implementation or just patch it? Otherwise we seem to run in similar issues like we had with our DDP and PyTorch's DDP where those implementations diverge over time.

Reply:
I don't like this fix personally, it was something thomas was hashing out earlier today. I'm going to revert this commit and keep this as an issue we need to fix upstream in the grad scaler or fairscale